pkg/controllers/user/logging/utils/fluentd.go

```diff
@@ -78,7 +77,9 @@ func (w *fluentForwarderTestWrap) sendData2Server(conn net.Conn, shareKey, usern
 		if _, err := conn.Write(fluentdForwarderTestData); err != nil {
 			return errors.Wrapf(err, "couldn't write data to fluentd forwarder %s", endpoint)
 		}
+		return nil
 	}
+
 	buf, err := readDataWithTimeout(conn)
```

Comment:
Can we add some comments here. I got confused by the reason why we read data from conn first.